ipa/ipa_ssh.py

```diff
@@ -0,0 +1,184 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+
+from ipa.ipa_exceptions import SSHProviderException
+from ipa.ipa_provider import IpaProvider
+
+
+class SSHProvider(IpaProvider):
+    """Class for testing instances in Azure."""
+
+    def __init__(self,
+                 access_key_id=None,  # Not used in SSH
+                 account_name=None,  # Not used in SSH
+                 cleanup=None,
+                 config=None,
+                 description=None,
+                 distro_name=None,
+                 early_exit=None,
+                 history_log=None,
+                 image_id=None,  # Not used in SSH
+                 inject=None,
+                 instance_type=None,  # Not used in SSH
+                 ip_address=None,
+                 log_level=30,
+                 no_default_test_dirs=False,
+                 provider_config=None,
+                 region=None,  # Not used in SSH
+                 results_dir=None,
+                 running_instance_id=None,  # Not used in SSH
+                 secret_access_key=None,  # Not used in SSH
+                 service_account_file=None,  # Not used in SSH
+                 ssh_key_name=None,  # Not used in SSH
+                 ssh_private_key=None,
+                 ssh_user=None,
+                 subnet_id=None,  # Not used in SSH
+                 test_dirs=None,
+                 test_files=None,
+                 timeout=None,
+                 vnet_name=None,  # Not used in SSH
+                 vnet_resource_group=None  # Not used in SSH
+                 ):
+        """Initialize Azure Provider class."""
+        super(SSHProvider, self).__init__('ssh',
+                                          cleanup,
+                                          config,
+                                          description,
+                                          distro_name,
+                                          early_exit,
+                                          history_log,
+                                          image_id,
+                                          inject,
+                                          instance_type,
+                                          log_level,
+                                          no_default_test_dirs,
+                                          provider_config,
+                                          region,
+                                          results_dir,
+                                          running_instance_id,
+                                          test_dirs,
+                                          test_files,
+                                          timeout)
+
+        # Cannot cleanup SSH instance
+        self.cleanup = False
+
+        self.ssh_private_key = (
+            ssh_private_key or
+            self._get_value(ssh_private_key, config_key='ssh_private_key')
+        )
```

Comment:
if os.path.isfile(self.ssh_private_key):
  self.ssh_private_key = self.ssh_private_key.read()

Reply:
See previous comment. The key file path is passed directly into SSH commands, not the value.